Automattic/jetpack
# Newsletter: Add "From" name customization

Fixes https://github.com/Automattic/jetpack/issues/37354

## Proposed changes:
* This PR adds the Jetpack side of the reply to name fields. Which will let users define what the newsletters name is set to by default this setting is set to the blog name or WordPress .com if the blogname is not set. 

New setting when
Default state
<img width="1074" alt="Screenshot 2024-05-15 at 1 29 00 PM" src="https://github.com/Automattic/jetpack/assets/115071/6536940e-e060-443a-8fda-114ef15c5be6">

my site is called Enej Dev jetpack. 

After typing in some content before save.
<img width="1072" alt="Screenshot 2024-05-15 at 1 29 36 PM" src="https://github.com/Automattic/jetpack/assets/115071/64c67e46-bfec-4132-882d-ca60e58277e1">



Related .com PR D148558-code

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:
1. load this branch 
2. Visit `/wp-admin/admin.php?page=jetpack&enable-newsletter-from-name=true#/newsletter` on your jetpack site. 
3. Save the option. (Notice that without `enable-newsletter-from-name=true` query parameter "feature" flag the option is not available. 
4. Notice that saving works and the option is still there as expected when you refresh the page.
